modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/display/context/ShippingMethodCheckoutStepDisplayContext.java

```diff
@@ -124,6 +166,42 @@ public String getCommerceShippingOptionLabel(
 			new CommerceShippingOptionLabelComparator());
 	}
 
+	public List<CommerceShippingOption> getCommerceShippingOptionsFiltered(
+			CommerceShippingMethod commerceShippingMethod)
+		throws PortalException {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)_httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		List<CommerceShippingOption> commerceShippingOptionsFiltered =
+			new ArrayList<>();
+		List<CommerceShippingOption> commerceShippingOptions =
+			getCommerceShippingOptions(commerceShippingMethod);
+		List<CommerceShippingFixedOption> commerceShippingFixedOptions =
+			getCommerceOrderFiltered();
+
+		for (CommerceShippingFixedOption commerceShippingFixedOption :
+				commerceShippingFixedOptions) {
+
+			for (CommerceShippingOption commerceShippingOption :
+					commerceShippingOptions) {
+
+				String name = commerceShippingFixedOption.getName(
+					themeDisplay.getLocale());
+				String key = commerceShippingFixedOption.getKey();
+
+				if (name.equals(commerceShippingOption.getLabel()) &&
```

Comment:
Should only validate the key not the name with the label, the name/label is a localized entry so it is dependant from the current locale. If the current session locale is different from the default this will fail.